x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValuesSourceReaderOperator.java

```diff
@@ -580,7 +580,7 @@ private ComputeBlockLoaderFactory(BlockFactory factory, int pageSize) {
 
         @Override
         public BlockLoader.BooleanBuilder booleansFromDocValues(int expectedCount) {
-            return factory.newBooleanBlockBuilder(expectedCount).mvOrdering(Block.MvOrdering.DEDUPLICATED_AND_SORTED_ASCENDING);
+            return factory.newBooleanBlockBuilder(expectedCount);
```

Comment:
@nik9000 it probably had a very good reason why we assumed this MvOrdering, but in practice unordered multi-values with dups, loaded from an index, ended up being marked as deduplicated which mv_dedupe used to optimize itself away into a no-op.

Reply:
Booleans come from https://lucene.apache.org/core/8_0_0/core/index.html?org/apache/lucene/document/SortedNumericDocValuesField.html which is sorted. They are stored as `0` and `1` and sorted. So all the `false` values should be before all of the `true` values.

So, I think, these should be `Block.MvOrdering.SORTED_ASCENDING`. Which a constant we don't have. FWIW it's much more efficient to deduplicate those. But we can get that code later.

I'm unsure how this made it this long. Numerics are not deduplicated - we can't pretend they are. Nor can we for booleans.